dotCMS/src/main/java/com/dotmarketing/portlets/personas/business/PersonaAPIImpl.java

```diff
@@ -34,381 +46,454 @@
+    }
+  }
+
+  @Override
+  public Host getParentHost(Persona persona) throws DotDataException, DotStateException, DotSecurityException {
+    return APILocator.getHostAPI().find(APILocator.getIdentifierAPI().find(persona).getHostId(), APILocator.getUserAPI().getSystemUser(),
+        false);
+  }
+
+  @Override
+  public List<Persona> getLiveHTMLPages(Host parent, User user, boolean respectFrontEndRoles)
+      throws DotDataException, DotSecurityException {
+    return getPersonas(parent, true, false, user, respectFrontEndRoles);
+  }
+
+  @Override
+  public List<Persona> getWorkingHTMLPages(Host parent, User user, boolean respectFrontEndRoles)
+      throws DotDataException, DotSecurityException {
+    return getPersonas(parent, false, false, user, respectFrontEndRoles);
+  }
+
+  @Override
+  public List<Persona> getDeletedPersonas(Host parent, User user, boolean respectFrontEndRoles)
+      throws DotDataException, DotSecurityException {
+    return getPersonas(parent, false, true, user, respectFrontEndRoles);
+  }
+
+  @Override
+  public List<Persona> getPersonas(Treeable parent, boolean live, boolean deleted, User user, boolean respectFrontEndRoles)
+      throws DotDataException, DotSecurityException {
+    return getPersonas(parent, live, deleted, -1, 0, "", user, respectFrontEndRoles);
+  }
+
+  @Override
+  public List<Persona> getPersonas(Treeable parent, boolean live, boolean deleted, int limit, int offset, String sortBy, User user,
+      boolean respectFrontEndRoles) throws DotDataException, DotSecurityException {
+    List<Persona> personas = new ArrayList<>();
+    StringBuilder query = new StringBuilder();
+    String liveWorkingDeleted = (live) ? " +live:true " : (deleted) ? " +working:true +deleted:true " : " +working:true -deleted:true ";
+    query.append(liveWorkingDeleted);
+    if (parent instanceof Host) {
+      query.append(" +conFolder:SYSTEM_FOLDER");
+      query.append(" +conHost:(").append(parent.getIdentifier()).append(" OR ").append(Host.SYSTEM_HOST).append(")");
+    } else if (parent instanceof Folder) {
+      query.append(" +conFolder:").append(parent.getIdentifier()).append(" ");
+    }
+
+    query.append(" +structureType:" + Structure.STRUCTURE_TYPE_PERSONA);
+    if (!UtilMethods.isSet(sortBy)) {
+      sortBy = "title desc";
+    }
+    List<Contentlet> contentlets =
+        APILocator.getContentletAPI().search(query.toString(), limit, offset, sortBy, user, respectFrontEndRoles);
+    for (Contentlet cont : contentlets) {
+      personas.add(fromContentlet(cont));
+    }
+    return personas;
+  }
+
+  @Override
+  public Tuple2<List<Persona>, Integer> getPersonasIncludingDefaultPersona(Treeable parent, String filter, boolean live, final int limit,
+       int offset, String sortBy, User user, boolean respectFrontEndRoles) throws DotDataException, DotSecurityException {
+
+    List<Persona> personas = new ArrayList<>();
+    StringBuilder query = new StringBuilder();
```

Comment:
I would probably separated the query creation from this method in order to make it a bit smaller